cmd/snap-bootstrap/initramfs_mounts_state.go

```diff
@@ -53,7 +59,33 @@ func (mst *initramfsMountsState) ReadEssential(recoverySystem string, essentialT
 	}
 
 	perf := timings.New(nil)
-	return seed.ReadSystemEssential(boot.InitramfsUbuntuSeedDir, recoverySystem, essentialTypes, perf)
+
+	// get the current time to pass to ReadSystemEssentialAndBetterEarliestTime
+	// note that we trust the time we have from the system, because that time
+	// comes from either:
+	// * a RTC on the system that the kernel/systemd consulted and used to move
+	//   time forward
+	// * systemd using a built-in timestamp from the initrd which was stamped
+	//   when the initrd was built, giving a lower bound on the current time if
+	//   the RTC does not have a battery or is otherwise unreliable, etc.
+	now := timeNow()
+
+	model, snaps, newTrustedEarliestTime, err := seed.ReadSystemEssentialAndBetterEarliestTime(boot.InitramfsUbuntuSeedDir, recoverySystem, essentialTypes, now, perf)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// set the time on the system to move forward if it is in the future - never
+	// move the time backwards
+	if newTrustedEarliestTime.After(now) {
+		if err := osutilSetTime(newTrustedEarliestTime); err != nil {
```

Comment:
@dbungert this what we do inside initrd on core systems. get all snaps verified and get the highest timestamp from all the verified assertions. And if that is newer than system jump, go back to the future.

imho `apt update --set-system-time` should be a thing.

cc: @julian-klode